src/main/kotlin/com/exactpro/th2/rptdataprovider/handlers/messages/MessageContinuousStream.kt

```diff
@@ -138,14 +138,14 @@ class MessageContinuousStream(
     }
 
     private suspend fun loadMoreMessage(): List<MessageBatchWrapper> {
-        if (lastElement == null || messageLoader == null) {
+        if (messageLoader == null) {
             return emptyList()
         }
 
         val messages = if (lastElement != null) {
             messageLoader!!.pullMoreMessageById(lastElement!!, firstPull, perStreamLimit)
         } else {
-            messageLoader!!.pullMoreMessageById(lastElement!!, firstPull, perStreamLimit)
+            messageLoader!!.pullMoreMessageByTimestamp(perStreamLimit)
         }

```

Comment:
Just to make it more Kotlinish
```kotlin
        val messages = messageLoader?.run {
           lastElement?.let {
             pullMoreMessageById(it, firstPull, perStreamLimit)
           } ?: pullMoreMessageByTimestamp(perStreamLimit)
        } ?: return emptyList()
```
Or instead of using `run` and `let` functions you can store values in local variables and check them for `null`. You won't need to use `!!` after that

I insist on one of those options because if the code was written correctly the `else` branch would be redundant (we checked `lastElement` for `null` and it cannot be `null` there)